lib/Doctrine/MongoDB/Aggregation/Stage/Operator.php

```diff
@@ -0,0 +1,988 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\MongoDB\Aggregation\Stage;
+
+use Doctrine\MongoDB\Aggregation\Builder;
+use Doctrine\MongoDB\Aggregation\Expr;
+use Doctrine\MongoDB\Aggregation\Stage;
+
+/**
+ * Fluent interface for adding operators to aggregation stages.
+ *
+ * @author alcaeus <alcaeus@alcaeus.org>
+ */
+abstract class Operator extends Stage
+{
+    /**
+     * @var Expr
+     */
+    protected $expr;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct(Builder $builder)
+    {
+        $this->expr = new Expr();
+
+        parent::__construct($builder);
+    }
+
+    /**
+     * Adds numbers together or adds numbers and a date. If one of the arguments is a date, $add treats the other arguments as milliseconds to add to the date.
+     *
+     * The arguments can be any valid expression as long as they resolve to either all numbers or to numbers and a date.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/add/
+     * @see Expr::add
+     * @param mixed|Expr $expression1
+     * @param mixed|Expr $expression2
+     * @param mixed|Expr $expression3,... Additional expressions
+     * @return self
+     */
+    public function add($expression1, $expression2 /* , $expression3, ... */)
```

Comment:
I believe you mentioned in IRC that `@method` would have shortcomings for some of our variadic methods and users not being able to get the appropriate error or IDE feedback. It makes sense to keep things as they are and wait until we can make use of traits down the line.